code/modules/power/groundmap_geothermal.dm

```diff
@@ -589,6 +598,64 @@ GLOBAL_VAR_INIT(active_bluespace_generators, 0)
 	icon_state = "circ-on75-cold"
 
 
+/// Psychic mist -- Spawns on generator if it explodes; provides point gen as the generator would have
+/obj/machinery/mist_origin
+	name = ""
+	resistance_flags = RESIST_ALL|PROJECTILE_IMMUNE|DROPSHIP_IMMUNE
+	//Lists all visual mist effects
+	var/list/mist_list = list()
+	//What hive we should add psy points to
+	var/hivenumber
+
+/obj/machinery/mist_origin/Initialize(mapload, last_corrupted_hivenumber)
+	. = ..()
+	hivenumber = last_corrupted_hivenumber
+	for(var/turf/tile in filled_circle_turfs(src, 10))
+		var/obj/effect/psychic_mist/new_mist = new(tile)
+		mist_list += new_mist
+	start_processing()
+
+// Mimic generator point gen
+/obj/machinery/mist_origin/process()
+	if(!(length(GLOB.humans_by_zlevel["2"]) > 0.2 * length(GLOB.alive_human_list_faction[FACTION_TERRAGOV])))
+		return
+	if(GLOB.total_bluespace_generators) //Avoid dividing by 0
+		var/points_generated = GENERATOR_PSYCH_POINT_OUTPUT / GLOB.total_bluespace_generators
+		SSpoints.add_strategic_psy_points(hivenumber, points_generated)
+		SSpoints.add_tactical_psy_points(hivenumber, points_generated*0.25)
+
+/obj/machinery/mist_origin/Destroy()
+	GLOB.total_bluespace_generators-- //Remove bluespace generator from psy-gen equation, since we're no longer producing points
+	for(var/obj/effect/psychic_mist/mist AS in mist_list)
+		QDEL_NULL(mist)
+	return ..()
+
+
+/obj/effect/psychic_mist
+	name = "psychic mist"
+	desc = "Condensed droplets of raw psychic energy swirl around you."
+	resistance_flags = RESIST_ALL|PROJECTILE_IMMUNE|DROPSHIP_IMMUNE
+	icon = 'icons/effects/weather_effects.dmi'
+	icon_state = "light_ash"
+	color = "#7f16c5"
+
+/obj/effect/psychic_mist/Initialize(mapload)
+	. = ..()
+	var/static/list/connections = list(
+		COMSIG_ATOM_ENTERED = PROC_REF(on_cross),
+	)
+	AddElement(/datum/element/connect_loc, connections)
+
+/// Psychic mist is difficult to breathe in, even with a mask on
```

Comment:
this is uh, incorrect isn't it? You are early returning if they have smoke protection, so a mask does in fact completely negate this.

Reply:
I don't think so
![image](https://github.com/user-attachments/assets/4dc87d2d-78d3-4fc1-a02e-05334cd4ec50)

Reply:
how's that show I am wrong? If they don't have smoke protection, they inhale smoke

Reply:
it's a misleading proc name
masks come under internals
![image](https://github.com/user-attachments/assets/5a086350-071a-4dd1-a5fd-99797f0f6e36)

Reply:
hold on my tired brain was trying to prove the exact opposite for some reason
yea good catch i'll change it